crates/typst/src/layout/inline/mod.rs

```diff
@@ -450,10 +450,10 @@ fn collect<'a>(
             let prev = full.len();
             let dir = TextElem::dir_in(styles);
             if dir != outer_dir {
-                // Insert "Explicit Directional Isolate".
+                // Insert "Explicit Directional Embedding".
```

Comment:
According to [Wikipedia](https://en.wikipedia.org/wiki/Bidirectional_text), the Embedding codepoints are legacy and discouraged in favor of the Isolates. Do you know what the concrete difference in semantics is?

Perhaps, we should continue to use the isolates, but apply them on consecutive runs?

Reply:
In terms of the difference between them, I got all my info from the spec

> Although the term embedding is used for some explicit formatting characters, the text within the scope of the embedding formatting characters is not independent of the surrounding text. Characters within an embedding can affect the ordering of characters outside, and vice versa. This is not the case with the isolate formatting characters, however. Characters within an isolate cannot affect the ordering of characters outside it, or vice versa. The effect that an isolate as a whole has on the ordering of the surrounding characters is the same as that of a neutral character, whereas an embedding or override roughly has the effect of a strong character.

https://unicode.org/reports/tr9/#Directional_Formatting_Characters

hmm didn't know they were discouraged will have a go using isolates for consecutive runs :+1:

Reply:
Had a go at finding consecutive runs, but it really overcomplicates things IMO. You can't just say if the text direction is different then previously I'll add an isolate / a pop, because other elements (especially space) that happen between the text direction change (i.e. after the last RTL but before the next LTR) should not be part of the isolated area, so you have to peek ahead to the next children.

I think that even though they are "discouraged" they are exactly what we need here, otherwise we're pretty much re-implementing parts of the Unicode BIDI algorithm ourselves...

Reply:
Ok, let's proceed with this for now. I will think it through again when I'm working on paragraph layout related things in the future.